The grep didn't match the OpenJDK java -version output.

Altough OpenJDK is not supported by Atlassian it eases use
to just run a quick test.

--- a/bin/check-java.sh	2018-02-20 11:54:30.000000000 +0100
+++ b/bin/check-java.sh	2018-03-03 13:16:48.259976953 +0100
@@ -7,7 +7,7 @@
 # we expect first line to be in format 'java version "1.8.0_40"' and assert that minor version number will be 8 or higher
 #
 
-"$_RUNJAVA" -version 2>&1 | grep "java version" | (
+"$_RUNJAVA" -version 2>&1 | grep " version" | (
         IFS=. read ignore1 version ignore2
         if [ ! ${version:-0} -ge "$_EXPECTED_JAVA_VERSION" ]
         then
